basedtyping/transformer.py

```diff
@@ -0,0 +1,262 @@
+"""utilities to create standard compatible annotations"""
+
+from __future__ import annotations
+
+import ast
+import sys
+import types
+import typing
+from contextlib import contextmanager
+from enum import Enum
+from functools import partial
+from typing import cast
+
+import typing_extensions
+from typing_extensions import override
+
+import basedtyping
+
+
+# ruff: noqa: S101
+class CringeTransformer(ast.NodeTransformer):
+    """Transforms `1 | 2` into `Literal[1] | Literal[2]` etc"""
+
+    def __init__(
+        self,
+        globalns: dict[str, object] | None,
+        localns: dict[str, object] | None,
+        *,
+        string_literals: bool,
+    ):
+        self.string_literals = string_literals
+
+        # This logic for handling Nones is copied from typing.ForwardRef._evaluate
+        if globalns is None and localns is None:
+            globalns = localns = {}
+        elif globalns is None:
+            assert localns is not None
+            globalns = localns
+        elif localns is None:
+            assert globalns is not None
+            localns = globalns
+
+        fair_and_unique_uuid_roll = "c4357574960843a2a8f9eb0c11aa88e5"
+        self.typing_name = f"_typing_extensions_{fair_and_unique_uuid_roll}"
+        self.basedtyping_name = f"_basedtyping_{fair_and_unique_uuid_roll}"
+        self.globalns = globalns
+
+        self.localns = localns | {
+            self.typing_name: typing_extensions,
+            self.basedtyping_name: basedtyping,
+        }
```

Comment:
`|` syntax is so stupid on dicts
```suggestion
        self.localns.extend({
            self.typing_name: typing_extensions,
            self.basedtyping_name: basedtyping,
        })
```

Reply:
why 🤔?

Reply:
to me it looks like a union of those two dicts (its either one or the other) when in reality its actually more like an intersection because its combining them. if anything it should use `&` syntax

but i think since the `extend` method exists which makes it more readable regardless the operator syntax should be avoided. imo overusing them makes code less readable